pootle/apps/pootle_store/diff.py

```diff
@@ -0,0 +1,226 @@
+                    next_index = next['index']
+                    delta = j2 - j1
+                    update_index_delta = (
+                        delta - next_index + previous_index + 1)
+
+                inserts.append((previous_index,
+                                new_unitid_list[j1:j2],
+                                next_index,
+                                update_index_delta))
+
+            elif tag == 'replace':
+                i1_index = self.db_units[self.active_units[i1 - 1]]['index']
+                i2_index = self.db_units[self.active_units[i2 - 1]]['index']
+                update_index_delta = j2 - j1 - i2_index + i1_index
+                inserts.append((i1_index,
+                                new_unitid_list[j1:j2],
+                                i2_index,
+                                update_index_delta))
+        return inserts
+
+    @cached_property
+    def new_unit_list(self):
+        # If file_revision is gte than the db_revision then new unit list
+        # will be exactly what is in the file
+        if self.file_revision >= self.db_revision:
+            return self.file_unit_ids
+
+        # These units are kept as they have been updated since file_revision
+        # but do not appear in the file
+        new_units = [u for u in self.updated_db_units
+                     if u not in self.file_unit_ids]
+
+        # These unit are either present in both or only in the file so are
+        # kept in the file order
+        new_units += [u for u in self.file_unit_ids
+                      if u not in self.obsoleted_db_units]
+
+        return new_units
+
+    @cached_property
+    def obsoleted_db_units(self):
+        return [unitid for unitid, unit in self.db_units.items()
+                if unit['state'] == OBSOLETE
+                and unit["revision"] > self.file_revision]
+
+    @cached_property
+    def opcodes(self):
+        sm = difflib.SequenceMatcher(None, self.active_units,
+                                     self.new_unit_list)
+        return sm.get_opcodes()
+
+    @cached_property
+    def updated_db_units(self):
+        return [unitid for unitid, unit in self.db_units.items()
+                if unit['revision'] > self.file_revision
+                and not unit["state"] == OBSOLETE]
+
+    def get_updated_dbids(self):
+        """Returns a set of unit DB ids to be updated.
+        :return: a set of unit DB ids to be updated.
+        """
+        update_dbids = set()
+
+        def _has_changed(uid):
+            """ check if the source and targets match
+            or are equal to empty string
+            """
+            file_unit = self.file_units[uid]
+            db_unit = self.db_units[uid]
+
+            db_source = multistring_to_python(db_unit["source_f"])
+            db_target = multistring_to_python(db_unit["target_f"])
+
+            source_same = (
+                (file_unit["source"] == ""
+                 and db_source == "")
```

Comment:
As already outlined in the past, [wrap conditionals expressions around the operators](http://docs.translatehouse.org/projects/translate-toolkit/en/latest/developers/styleguide.html#if-statements).